app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -201,18 +218,38 @@ export default class SchoolDisciplineDashboard extends React.Component {
     });
   }
 
-  allIncidentTypes(students) {
-    const incidents = students.map(student => {
-      return student.discipline_incidents ? student.discipline_incidents : [];
-    });
-    return _.uniqBy(_.flatten(incidents), 'incident_code')
-    .map(incident => incident.incident_code);
+  allIncidentTypes() {
+    const {dashboardStudents} = this.props;
+    const incidents = _.uniqBy(_.compact(_.flatten(dashboardStudents.map(student => student.discipline_incidents))), 'incident_code');
+    return incidents.map(incident => incident.incident_code).sort();
```

Comment:
As a small style suggestions, if this is just trying to get all the codes, maybe flat map then uniq then sort (like you do in the other methods)?  It doesn't seem like the whole object of `incidents` is used.

Reply:
This was an attempt to make it more readable, since this one has to iterate through students and then incidents. Are you saying it would be preferable to have it all on one line?